src/resources/projects.js

```diff
@@ -0,0 +1,77 @@
+import fs from 'fs';
+import express from 'express';
+
+const router = express.Router();
+const projects = require('../data/projects.json');
+
+// update project
+router.put('/edit/:id', (req, res) => {
+  const upProject = req.body;
+  projects.forEach((project) => {
+    if (parseInt(project.id, 10) === parseInt(req.params.id, 10)) {
+      /* edit the name and the start_date of a project */
+      const proj = project; // this works because it is passed always by reference in js
+      proj.project_name = upProject.project_name ? upProject.project_name : project.project_name;
```

Comment:
This can be replaced by 
```suggestion
       proj.project_name = upProject.project_name ?? project.project_name;
```

The nullish coalescing operator (??) uses the value of the left side of the expression if the left side of the expression is not null nor undefined, if it is null or undefined the value of the right side of the expression is used.

Same comment applies to the following lines